sdk/appconfiguration/azure-appconfigurationprovider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,219 @@
+                    provider._dict[trimmed_key] = secret
+                elif provider.__is_json_content_type(config.content_type):
+                    try:
+                        j_object = json.loads(config.value)
+                        provider._dict[trimmed_key] = j_object
+                    except json.JSONDecodeError as e:
+                        # If the value is not a valid JSON, treat it like regular string value
+                        provider._dict[trimmed_key] = config.value
+                else:
+                    provider._dict[trimmed_key] = config.value
+        return provider
+
+    def __buildprovider(self, connection_string, endpoint, credential, key_vault_options):
+        headers = {}
+        correlation_context = "RequestType=Startup"
+
+        if (key_vault_options and
+                (key_vault_options.credential or
+                key_vault_options.secret_clients or key_vault_options.secret_resolver)):
+            correlation_context += ",UsesKeyVault"
+
+        headers["Correlation-Context"] = correlation_context
+        useragent = USER_AGENT
+
+        if (connection_string and endpoint):
+            raise AttributeError(
+                "Both connection_string and endpoint are set. Only one of these should be set.")
+
+        if connection_string:
+            self._client = AzureAppConfigurationClient.from_connection_string(
+                connection_string, user_agent=useragent, headers=headers)
+            return
+        self._client = AzureAppConfigurationClient(
+            endpoint, credential, user_agent=useragent, headers=headers)
+
+    @staticmethod
+    def __resolve_keyvault_reference(config, key_vault_options, secret_clients):
+        if key_vault_options is None:
+            raise AttributeError(
+                "Key Vault options must be set to resolve Key Vault references.")
+
+        if config.secret_id is None:
+            raise AttributeError(
+                "Key Vault reference must have a uri value.")
+
+        key_vault_identifier = KeyVaultSecretIdentifier(config.secret_id)
+
+        referenced_client = next(
+            (client for client in secret_clients if client.vault_url == key_vault_identifier.vault_url), None)
+
+        if referenced_client is None and key_vault_options.credential is not None:
+            referenced_client = SecretClient(
+                vault_url=key_vault_identifier.vault_url, credential=key_vault_options.credential)
+            secret_clients[key_vault_identifier.vault_url] = referenced_client
+
+        if referenced_client:
+            try:
+                return referenced_client.get_secret(key_vault_identifier.name, version=key_vault_identifier.version)
+            except ResourceNotFoundError as e:
+                raise KeyVaultReferenceError("Key Vault %s does not contain secret %s" % (
+                    key_vault_identifier.vault_url, key_vault_identifier.name))
+            except HttpResponseError as e:
```

Comment:
Same here, we can remove HttpResponseError exception catching and handle specific errors like 401/403 - and throw them as KeyVaultReferenceError